### PR TITLE
 [Scale transform] Enhancements

### PR DESCRIPTION
**Pull request by clinique**
* Adding 'NaN' key to nicely take care of non numeric values presented to scale transform
 * Adding the ability to format result presented with a format string provided in the .scale file.
 * A bit of code cleansing

If this PR finds an agreement, I'll update Scale transformation documentation accordingly

Signed-off-by: Gaël L'hopital <gael@lhopital.org>

### CODE REVIEW

File: bundles/org.openhab.transform.scale/src/main/java/org/openhab/transform/scale/internal/ScaleTransformationService.java [hunk 1]
```diff
@@ -125,24 +147,29 @@ private String getScaleResult(Map<Range, String> data, String source, final BigD
...
+                    final boolean upperInclusive = matcher.group(4).equals("]");
 
                     final String lowLimit = matcher.group(2);
                     final String highLimit = matcher.group(3);
 
                     try {
-                        final BigDecimal lowValue = lowLimit.isEmpty() ? null : new BigDecimal(lowLimit);
-                        final BigDecimal highValue = highLimit.isEmpty() ? null : new BigDecimal(highLimit);
-                        final Range range = Range.range(lowValue, lowerInclusive, highValue, upperInclusive);
+                        final Range range = Range.range(lowLimit.isEmpty() ? null : new BigDecimal(lowLimit),
```
Comment (J-N-K) [inline bundles/org.openhab.transform.scale/src/main/java/org/openhab/transform/scale/internal/ScaleTransformationService.java]: That's fully correct code-wise, but hard to read. would you mind reverting that?

File: bundles/org.openhab.transform.scale/src/main/java/org/openhab/transform/scale/internal/ScaleTransformationService.java [hunk 2]
```diff
@@ -125,24 +147,29 @@ private String getScaleResult(Map<Range, String> data, String source, final BigD
                 final String value = properties.getProperty(entry);
                 final Matcher matcher = LIMITS_PATTERN.matcher(entry);
                 if (matcher.matches() && (matcher.groupCount() == 4)) {
-                    final boolean lowerInclusive = matcher.group(1).equals("]") ? false : true;
-                    final boolean upperInclusive = matcher.group(4).equals("[") ? false : true;
+                    final boolean lowerInclusive = matcher.group(1).equals("[");
```
Comment (J-N-K) [inline bundles/org.openhab.transform.scale/src/main/java/org/openhab/transform/scale/internal/ScaleTransformationService.java]: nice

File: bundles/org.openhab.transform.scale/conf/transform/scale/catchnonnumeric.scale [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[-8..10[=low
+[10..20[=middle
+[20..300[=high
+[300..]=extreme
+non-number=Non Numeric
```
Comment (J-N-K) [inline bundles/org.openhab.transform.scale/conf/transform/scale/catchnonnumeric.scale]: I like the idea but how can this happen?

Comment (clinique) [reply bundles/org.openhab.transform.scale/conf/transform/scale/catchnonnumeric.scale]: Yes, this happens when values changes to NULL or UNDEF, to log file is plenty of warnings stating that Scale can only be applied to numeric values. In this case, if the user decides it, no warning and a nice output.

Comment (J-N-K) [reply bundles/org.openhab.transform.scale/conf/transform/scale/catchnonnumeric.scale]: I'm not so sure about the naming, though. I would prefer `invalid`, `NaN`, `undefined` or something like that. Or: can we re-use `[..]` for undefined ranges for that?

**Comment by J-N-K** [comment]
Documentation is still missing.

**Comment by clinique** [comment]
PR pushed [here](https://github.com/eclipse/smarthome/pull/6896)

**Comment by J-N-K** [comment]
The additions itself are ok but need to be applied to [this file](https://github.com/openhab/openhab2-addons/tree/master/bundles/org.openhab.transform.scale) in this repo.

**Comment by J-N-K** [comment]
You forgot to sign-off your last commit,  can you fix that?

**Comment by clinique** [comment]
> You forgot to sign-off your last commit, can you fix that?

I had to directly edit the README.MD file on github because for an unknown reason git push did not make it. Will try to sign now :(

**Comment by clinique** [comment]
> > You forgot to sign-off your last commit, can you fix that?
> 
> I had to directly edit the README.MD file on github because for an unknown reason git push did not make it. Will try to sign now :(

Too tired, I was not editing the good branch. Git is always right !

**Pull request merged**